{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-11-28T22:26:53.069195Z",
     "start_time": "2023-11-28T22:26:53.064499Z"
    }
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-11-28T22:26:53.314294Z",
     "start_time": "2023-11-28T22:26:53.308447Z"
    }
   },
   "outputs": [],
   "source": [
    "mainindir=\"/scratch/Shares/dowell/dbnascent/out/meta_analysis/snpconnecter/outputdata/\"\n",
    "rootname=\"EFO_0000565\"\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Read in all the data that came from the GWAStogenes.sh script "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-11-28T22:26:53.981181Z",
     "start_time": "2023-11-28T22:26:53.718147Z"
    }
   },
   "outputs": [],
   "source": [
    "def snpclosestgenedf(mainindir, rootname, tissue):\n",
    "    snptobedfile=mainindir+rootname+\"_\"+tissue+\"/\"+rootname+\".snp_to_bed.txt\"\n",
    "    snptobeddf = pd.read_csv(snptobedfile, sep=\"\\t\")\n",
    "    closestgenetosnp=mainindir+rootname+\"_\"+tissue+\"/\"+rootname+\".closestgene.sorted.bed\"\n",
    "    closestgenetosnpdf = pd.read_csv(closestgenetosnp, sep=\"\\t\", names=[\"snp_chr\", \"snp_start\", \"snp_stop\", \"closest_gene_chr\", \"closest_gene_start\", \"closest_gene_stop\", \"closest_gene_name\",\"score\", \"strand\", \"closest_distance\"], index_col=False)\n",
    "    closestgenetosnpdf[[\"closest_gene\", \"closest_transcript_id\"]] = closestgenetosnpdf['closest_gene_name'].str.split(':', n=1, expand=True)\n",
    "    masterdf = snptobeddf.merge(closestgenetosnpdf, on=[\"snp_chr\", \"snp_start\"])\n",
    "    print (\"unique snps in bidirectionals\", masterdf[\"rsID\"].nunique())\n",
    "    return masterdf\n",
    "    \n",
    "\n",
    "def merge_dfs_outer(mainindir, rootname, tissue):\n",
    "    snptobedfile=mainindir+rootname+\"_\"+tissue+\"/\"+rootname+\".snp_to_bed.txt\"\n",
    "    bedtobidirfile=mainindir+rootname+\"_\"+tissue+\"/\"+rootname+\".snpinbidir.df\"\n",
    "    bidirtogenefile=mainindir+rootname+\"_\"+tissue+\"/\"+rootname+\".genelist.txt.df\"\n",
    "    snptobeddf = pd.read_csv(snptobedfile, sep=\"\\t\")\n",
    "    checkedbidirs = readsnpsandgeneschecked()\n",
    "    bedtobidirdf = pd.read_csv(bedtobidirfile)\n",
    "    bidirtogenedf = pd.read_csv(bidirtogenefile)\n",
    "    bedtobidirdf[\"bidirtesed_for_links\"]= bedtobidirdf[\"bidirname\"].isin(checkedbidirs)\n",
    "    masterdf = snptobeddf.merge(bedtobidirdf, on=[\"snp_chr\", \"snp_start\"], how=\"outer\")\n",
    "    masterdf = masterdf.merge(bidirtogenedf, left_on=\"bidirname\", right_on=\"transcript_2\", how=\"outer\")\n",
    "    masterdf_mini = masterdf[[\"riskAllele\", \"rsID\", 'snp_chr', 'snp_start', \"bidirname\", \"distance_tss\", \"transcript_1\", \"gene\", \"bidirtesed_for_links\"]].drop_duplicates()\n",
    "    closestgenetosnpdf = snpclosestgenedf(mainindir, rootname, tissue)\n",
    "    backtoallsnps = closestgenetosnpdf.merge(masterdf_mini, how=\"outer\", on=[\"snp_chr\", \"snp_start\", \"riskAllele\", \"rsID\"])\n",
    "    backtoallsnps[\"snp_in_gene\"] = backtoallsnps[\"closest_distance\"]==0\n",
    "    return backtoallsnps\n",
    "\n",
    "#\n",
    "def readsnpsandgeneschecked():\n",
    "    #only genes and snps with expression were checked for correlation\n",
    "    filename=\"/scratch/Shares/dowell/rutendo/projects/DBNascent_Analysis/data/counts/normalized/genes_inter_bidir_filtered.tsv.gz\"\n",
    "    checkedforlinksdf = pd.read_csv(filename, compression='gzip', sep=\"\\t\",low_memory=False)\n",
    "    bidircheckedforlinksdf = checkedforlinksdf[checkedforlinksdf[\"gene_transcript\"].str.startswith(\"chr\")]\n",
    "    return bidircheckedforlinksdf[\"gene_transcript\"].unique()\n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-11-28T22:29:05.069486Z",
     "start_time": "2023-11-28T22:26:53.987440Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "unique snps in bidirectionals 7805\n"
     ]
    }
   ],
   "source": [
    "tissue=\"blood\"\n",
    "allsnps_one_tissue = merge_dfs_outer(mainindir, rootname, tissue)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-11-28T22:29:05.239806Z",
     "start_time": "2023-11-28T22:29:05.074637Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>riskAllele</th>\n",
       "      <th>locations</th>\n",
       "      <th>snp_chr</th>\n",
       "      <th>snp_start</th>\n",
       "      <th>rsID</th>\n",
       "      <th>change</th>\n",
       "      <th>stop</th>\n",
       "      <th>snp_stop</th>\n",
       "      <th>closest_gene_chr</th>\n",
       "      <th>...</th>\n",
       "      <th>strand</th>\n",
       "      <th>closest_distance</th>\n",
       "      <th>closest_gene</th>\n",
       "      <th>closest_transcript_id</th>\n",
       "      <th>bidirname</th>\n",
       "      <th>distance_tss</th>\n",
       "      <th>transcript_1</th>\n",
       "      <th>gene</th>\n",
       "      <th>bidirtesed_for_links</th>\n",
       "      <th>snp_in_gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>rs7090445-?</td>\n",
       "      <td>10:61961417</td>\n",
       "      <td>chr10</td>\n",
       "      <td>61961417</td>\n",
       "      <td>rs7090445</td>\n",
       "      <td>?</td>\n",
       "      <td>61961418</td>\n",
       "      <td>61961418</td>\n",
       "      <td>chr10</td>\n",
       "      <td>...</td>\n",
       "      <td>+</td>\n",
       "      <td>88689</td>\n",
       "      <td>ARID5B</td>\n",
       "      <td>NM_001244638.2</td>\n",
       "      <td>chr10:61961221-61961505</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>63</td>\n",
       "      <td>rs7090445-?</td>\n",
       "      <td>10:61961417</td>\n",
       "      <td>chr10</td>\n",
       "      <td>61961417</td>\n",
       "      <td>rs7090445</td>\n",
       "      <td>?</td>\n",
       "      <td>61961418</td>\n",
       "      <td>61961418</td>\n",
       "      <td>chr10</td>\n",
       "      <td>...</td>\n",
       "      <td>+</td>\n",
       "      <td>88689</td>\n",
       "      <td>ARID5B</td>\n",
       "      <td>NM_001244638.2</td>\n",
       "      <td>chr10:61961221-61961505</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>rs2665658-A</td>\n",
       "      <td>2:60599667</td>\n",
       "      <td>chr2</td>\n",
       "      <td>60599667</td>\n",
       "      <td>rs2665658</td>\n",
       "      <td>A</td>\n",
       "      <td>60599668</td>\n",
       "      <td>60599668</td>\n",
       "      <td>chr2</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>71883</td>\n",
       "      <td>BCL11A</td>\n",
       "      <td>NR_175964.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2</td>\n",
       "      <td>rs2665658-A</td>\n",
       "      <td>2:60599667</td>\n",
       "      <td>chr2</td>\n",
       "      <td>60599667</td>\n",
       "      <td>rs2665658</td>\n",
       "      <td>A</td>\n",
       "      <td>60599668</td>\n",
       "      <td>60599668</td>\n",
       "      <td>chr2</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>71883</td>\n",
       "      <td>BCL11A</td>\n",
       "      <td>NR_175964.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3</td>\n",
       "      <td>rs4869742-T</td>\n",
       "      <td>6:151586613</td>\n",
       "      <td>chr6</td>\n",
       "      <td>151586613</td>\n",
       "      <td>rs4869742</td>\n",
       "      <td>T</td>\n",
       "      <td>151586614</td>\n",
       "      <td>151586614</td>\n",
       "      <td>chr6</td>\n",
       "      <td>...</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>CCDC170</td>\n",
       "      <td>NM_025059.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13082</th>\n",
       "      <td>2110</td>\n",
       "      <td>rs131821-AT</td>\n",
       "      <td>22:50511648</td>\n",
       "      <td>chr22</td>\n",
       "      <td>50511648</td>\n",
       "      <td>rs131821</td>\n",
       "      <td>AT</td>\n",
       "      <td>50511649</td>\n",
       "      <td>50511649</td>\n",
       "      <td>chr22</td>\n",
       "      <td>...</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>NCAPH2</td>\n",
       "      <td>NM_014551.5</td>\n",
       "      <td>chr22:50511408-50511748</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13083</th>\n",
       "      <td>2111</td>\n",
       "      <td>rs11715604-?</td>\n",
       "      <td>3:136870707</td>\n",
       "      <td>chr3</td>\n",
       "      <td>136870707</td>\n",
       "      <td>rs11715604</td>\n",
       "      <td>?</td>\n",
       "      <td>136870708</td>\n",
       "      <td>136870708</td>\n",
       "      <td>chr3</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>9395</td>\n",
       "      <td>NCK1-DT</td>\n",
       "      <td>NR_110175.1</td>\n",
       "      <td>chr3:136869554-136871154</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13084</th>\n",
       "      <td>2112</td>\n",
       "      <td>rs9421684-A</td>\n",
       "      <td>10:80702590</td>\n",
       "      <td>chr10</td>\n",
       "      <td>80702590</td>\n",
       "      <td>rs9421684</td>\n",
       "      <td>A</td>\n",
       "      <td>80702591</td>\n",
       "      <td>80702591</td>\n",
       "      <td>chr10</td>\n",
       "      <td>...</td>\n",
       "      <td>+</td>\n",
       "      <td>48857</td>\n",
       "      <td>LINC02655</td>\n",
       "      <td>NR_155749.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13085</th>\n",
       "      <td>2113</td>\n",
       "      <td>rs11187157-C</td>\n",
       "      <td>10:92742487</td>\n",
       "      <td>chr10</td>\n",
       "      <td>92742487</td>\n",
       "      <td>rs11187157</td>\n",
       "      <td>C</td>\n",
       "      <td>92742488</td>\n",
       "      <td>92742488</td>\n",
       "      <td>chr10</td>\n",
       "      <td>...</td>\n",
       "      <td>+</td>\n",
       "      <td>46841</td>\n",
       "      <td>HHEX</td>\n",
       "      <td>NM_002729.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13086</th>\n",
       "      <td>2114</td>\n",
       "      <td>rs12203592-T</td>\n",
       "      <td>6:396321</td>\n",
       "      <td>chr6</td>\n",
       "      <td>396321</td>\n",
       "      <td>rs12203592</td>\n",
       "      <td>T</td>\n",
       "      <td>396322</td>\n",
       "      <td>396322</td>\n",
       "      <td>chr6</td>\n",
       "      <td>...</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>IRF4</td>\n",
       "      <td>NM_002460.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>13087 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Unnamed: 0    riskAllele    locations snp_chr  snp_start        rsID  \\\n",
       "0               0   rs7090445-?  10:61961417   chr10   61961417   rs7090445   \n",
       "1              63   rs7090445-?  10:61961417   chr10   61961417   rs7090445   \n",
       "2               1   rs2665658-A   2:60599667    chr2   60599667   rs2665658   \n",
       "3               2   rs2665658-A   2:60599667    chr2   60599667   rs2665658   \n",
       "4               3   rs4869742-T  6:151586613    chr6  151586613   rs4869742   \n",
       "...           ...           ...          ...     ...        ...         ...   \n",
       "13082        2110   rs131821-AT  22:50511648   chr22   50511648    rs131821   \n",
       "13083        2111  rs11715604-?  3:136870707    chr3  136870707  rs11715604   \n",
       "13084        2112   rs9421684-A  10:80702590   chr10   80702590   rs9421684   \n",
       "13085        2113  rs11187157-C  10:92742487   chr10   92742487  rs11187157   \n",
       "13086        2114  rs12203592-T     6:396321    chr6     396321  rs12203592   \n",
       "\n",
       "      change       stop   snp_stop closest_gene_chr  ...  strand  \\\n",
       "0          ?   61961418   61961418            chr10  ...       +   \n",
       "1          ?   61961418   61961418            chr10  ...       +   \n",
       "2          A   60599668   60599668             chr2  ...       -   \n",
       "3          A   60599668   60599668             chr2  ...       -   \n",
       "4          T  151586614  151586614             chr6  ...       +   \n",
       "...      ...        ...        ...              ...  ...     ...   \n",
       "13082     AT   50511649   50511649            chr22  ...       +   \n",
       "13083      ?  136870708  136870708             chr3  ...       -   \n",
       "13084      A   80702591   80702591            chr10  ...       +   \n",
       "13085      C   92742488   92742488            chr10  ...       +   \n",
       "13086      T     396322     396322             chr6  ...       +   \n",
       "\n",
       "       closest_distance closest_gene closest_transcript_id  \\\n",
       "0                 88689       ARID5B        NM_001244638.2   \n",
       "1                 88689       ARID5B        NM_001244638.2   \n",
       "2                 71883       BCL11A           NR_175964.1   \n",
       "3                 71883       BCL11A           NR_175964.1   \n",
       "4                     0      CCDC170           NM_025059.4   \n",
       "...                 ...          ...                   ...   \n",
       "13082                 0       NCAPH2           NM_014551.5   \n",
       "13083              9395      NCK1-DT           NR_110175.1   \n",
       "13084             48857    LINC02655           NR_155749.1   \n",
       "13085             46841         HHEX           NM_002729.5   \n",
       "13086                 0         IRF4           NM_002460.4   \n",
       "\n",
       "                      bidirname  distance_tss transcript_1 gene  \\\n",
       "0       chr10:61961221-61961505           NaN          NaN  NaN   \n",
       "1       chr10:61961221-61961505           NaN          NaN  NaN   \n",
       "2                           NaN           NaN          NaN  NaN   \n",
       "3                           NaN           NaN          NaN  NaN   \n",
       "4                           NaN           NaN          NaN  NaN   \n",
       "...                         ...           ...          ...  ...   \n",
       "13082   chr22:50511408-50511748           NaN          NaN  NaN   \n",
       "13083  chr3:136869554-136871154           NaN          NaN  NaN   \n",
       "13084                       NaN           NaN          NaN  NaN   \n",
       "13085                       NaN           NaN          NaN  NaN   \n",
       "13086                       NaN           NaN          NaN  NaN   \n",
       "\n",
       "      bidirtesed_for_links  snp_in_gene  \n",
       "0                    False        False  \n",
       "1                    False        False  \n",
       "2                      NaN        False  \n",
       "3                      NaN        False  \n",
       "4                      NaN         True  \n",
       "...                    ...          ...  \n",
       "13082                False         True  \n",
       "13083                False        False  \n",
       "13084                  NaN        False  \n",
       "13085                  NaN        False  \n",
       "13086                  NaN         True  \n",
       "\n",
       "[13087 rows x 24 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "allsnps_one_tissue"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-11-28T22:29:05.377517Z",
     "start_time": "2023-11-28T22:29:05.243946Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "unique snps in genes 3245\n",
      "unique snps not in genes 4560\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>riskAllele</th>\n",
       "      <th>locations</th>\n",
       "      <th>snp_chr</th>\n",
       "      <th>snp_start</th>\n",
       "      <th>rsID</th>\n",
       "      <th>change</th>\n",
       "      <th>stop</th>\n",
       "      <th>snp_stop</th>\n",
       "      <th>closest_gene_chr</th>\n",
       "      <th>...</th>\n",
       "      <th>strand</th>\n",
       "      <th>closest_distance</th>\n",
       "      <th>closest_gene</th>\n",
       "      <th>closest_transcript_id</th>\n",
       "      <th>bidirname</th>\n",
       "      <th>distance_tss</th>\n",
       "      <th>transcript_1</th>\n",
       "      <th>gene</th>\n",
       "      <th>bidirtesed_for_links</th>\n",
       "      <th>snp_in_gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3</td>\n",
       "      <td>rs4869742-T</td>\n",
       "      <td>6:151586613</td>\n",
       "      <td>chr6</td>\n",
       "      <td>151586613</td>\n",
       "      <td>rs4869742</td>\n",
       "      <td>T</td>\n",
       "      <td>151586614</td>\n",
       "      <td>151586614</td>\n",
       "      <td>chr6</td>\n",
       "      <td>...</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>CCDC170</td>\n",
       "      <td>NM_025059.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>6</td>\n",
       "      <td>rs72832868-A</td>\n",
       "      <td>2:110850099</td>\n",
       "      <td>chr2</td>\n",
       "      <td>110850099</td>\n",
       "      <td>rs72832868</td>\n",
       "      <td>A</td>\n",
       "      <td>110850100</td>\n",
       "      <td>110850100</td>\n",
       "      <td>chr2</td>\n",
       "      <td>...</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>ACOXL</td>\n",
       "      <td>NM_001365487.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>7</td>\n",
       "      <td>rs114959229-T</td>\n",
       "      <td>22:46469808</td>\n",
       "      <td>chr22</td>\n",
       "      <td>46469808</td>\n",
       "      <td>rs114959229</td>\n",
       "      <td>T</td>\n",
       "      <td>46469809</td>\n",
       "      <td>46469809</td>\n",
       "      <td>chr22</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>0</td>\n",
       "      <td>CELSR1</td>\n",
       "      <td>NM_014246.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>8</td>\n",
       "      <td>rs139440276-T</td>\n",
       "      <td>3:66312665</td>\n",
       "      <td>chr3</td>\n",
       "      <td>66312665</td>\n",
       "      <td>rs139440276</td>\n",
       "      <td>T</td>\n",
       "      <td>66312666</td>\n",
       "      <td>66312666</td>\n",
       "      <td>chr3</td>\n",
       "      <td>...</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>SLC25A26</td>\n",
       "      <td>NM_001379210.1</td>\n",
       "      <td>chr3:66312484-66313196</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>10</td>\n",
       "      <td>rs59382229-G</td>\n",
       "      <td>1:241338685</td>\n",
       "      <td>chr1</td>\n",
       "      <td>241338685</td>\n",
       "      <td>rs59382229</td>\n",
       "      <td>G</td>\n",
       "      <td>241338686</td>\n",
       "      <td>241338686</td>\n",
       "      <td>chr1</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>0</td>\n",
       "      <td>RGS7</td>\n",
       "      <td>NM_001350116.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13079</th>\n",
       "      <td>2107</td>\n",
       "      <td>rs181181503-C</td>\n",
       "      <td>12:74276187</td>\n",
       "      <td>chr12</td>\n",
       "      <td>74276187</td>\n",
       "      <td>rs181181503</td>\n",
       "      <td>C</td>\n",
       "      <td>74276188</td>\n",
       "      <td>74276188</td>\n",
       "      <td>chr12</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>0</td>\n",
       "      <td>LINC02882</td>\n",
       "      <td>NR_038300.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13080</th>\n",
       "      <td>2108</td>\n",
       "      <td>rs1576377-C</td>\n",
       "      <td>13:60686390</td>\n",
       "      <td>chr13</td>\n",
       "      <td>60686390</td>\n",
       "      <td>rs1576377</td>\n",
       "      <td>C</td>\n",
       "      <td>60686391</td>\n",
       "      <td>60686391</td>\n",
       "      <td>chr13</td>\n",
       "      <td>...</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>LINC00378</td>\n",
       "      <td>NR_047003.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13081</th>\n",
       "      <td>2109</td>\n",
       "      <td>rs73005220-G</td>\n",
       "      <td>19:16161878</td>\n",
       "      <td>chr19</td>\n",
       "      <td>16161878</td>\n",
       "      <td>rs73005220</td>\n",
       "      <td>G</td>\n",
       "      <td>16161879</td>\n",
       "      <td>16161879</td>\n",
       "      <td>chr19</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>0</td>\n",
       "      <td>CIB3</td>\n",
       "      <td>NM_054113.4</td>\n",
       "      <td>chr19:16161672-16162168</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13082</th>\n",
       "      <td>2110</td>\n",
       "      <td>rs131821-AT</td>\n",
       "      <td>22:50511648</td>\n",
       "      <td>chr22</td>\n",
       "      <td>50511648</td>\n",
       "      <td>rs131821</td>\n",
       "      <td>AT</td>\n",
       "      <td>50511649</td>\n",
       "      <td>50511649</td>\n",
       "      <td>chr22</td>\n",
       "      <td>...</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>NCAPH2</td>\n",
       "      <td>NM_014551.5</td>\n",
       "      <td>chr22:50511408-50511748</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13086</th>\n",
       "      <td>2114</td>\n",
       "      <td>rs12203592-T</td>\n",
       "      <td>6:396321</td>\n",
       "      <td>chr6</td>\n",
       "      <td>396321</td>\n",
       "      <td>rs12203592</td>\n",
       "      <td>T</td>\n",
       "      <td>396322</td>\n",
       "      <td>396322</td>\n",
       "      <td>chr6</td>\n",
       "      <td>...</td>\n",
       "      <td>+</td>\n",
       "      <td>0</td>\n",
       "      <td>IRF4</td>\n",
       "      <td>NM_002460.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5395 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Unnamed: 0     riskAllele    locations snp_chr  snp_start         rsID  \\\n",
       "4               3    rs4869742-T  6:151586613    chr6  151586613    rs4869742   \n",
       "7               6   rs72832868-A  2:110850099    chr2  110850099   rs72832868   \n",
       "8               7  rs114959229-T  22:46469808   chr22   46469808  rs114959229   \n",
       "9               8  rs139440276-T   3:66312665    chr3   66312665  rs139440276   \n",
       "11             10   rs59382229-G  1:241338685    chr1  241338685   rs59382229   \n",
       "...           ...            ...          ...     ...        ...          ...   \n",
       "13079        2107  rs181181503-C  12:74276187   chr12   74276187  rs181181503   \n",
       "13080        2108    rs1576377-C  13:60686390   chr13   60686390    rs1576377   \n",
       "13081        2109   rs73005220-G  19:16161878   chr19   16161878   rs73005220   \n",
       "13082        2110    rs131821-AT  22:50511648   chr22   50511648     rs131821   \n",
       "13086        2114   rs12203592-T     6:396321    chr6     396321   rs12203592   \n",
       "\n",
       "      change       stop   snp_stop closest_gene_chr  ...  strand  \\\n",
       "4          T  151586614  151586614             chr6  ...       +   \n",
       "7          A  110850100  110850100             chr2  ...       +   \n",
       "8          T   46469809   46469809            chr22  ...       -   \n",
       "9          T   66312666   66312666             chr3  ...       +   \n",
       "11         G  241338686  241338686             chr1  ...       -   \n",
       "...      ...        ...        ...              ...  ...     ...   \n",
       "13079      C   74276188   74276188            chr12  ...       -   \n",
       "13080      C   60686391   60686391            chr13  ...       +   \n",
       "13081      G   16161879   16161879            chr19  ...       -   \n",
       "13082     AT   50511649   50511649            chr22  ...       +   \n",
       "13086      T     396322     396322             chr6  ...       +   \n",
       "\n",
       "       closest_distance closest_gene closest_transcript_id  \\\n",
       "4                     0      CCDC170           NM_025059.4   \n",
       "7                     0        ACOXL        NM_001365487.2   \n",
       "8                     0       CELSR1           NM_014246.4   \n",
       "9                     0     SLC25A26        NM_001379210.1   \n",
       "11                    0         RGS7        NM_001350116.1   \n",
       "...                 ...          ...                   ...   \n",
       "13079                 0    LINC02882           NR_038300.1   \n",
       "13080                 0    LINC00378           NR_047003.1   \n",
       "13081                 0         CIB3           NM_054113.4   \n",
       "13082                 0       NCAPH2           NM_014551.5   \n",
       "13086                 0         IRF4           NM_002460.4   \n",
       "\n",
       "                     bidirname  distance_tss transcript_1 gene  \\\n",
       "4                          NaN           NaN          NaN  NaN   \n",
       "7                          NaN           NaN          NaN  NaN   \n",
       "8                          NaN           NaN          NaN  NaN   \n",
       "9       chr3:66312484-66313196           NaN          NaN  NaN   \n",
       "11                         NaN           NaN          NaN  NaN   \n",
       "...                        ...           ...          ...  ...   \n",
       "13079                      NaN           NaN          NaN  NaN   \n",
       "13080                      NaN           NaN          NaN  NaN   \n",
       "13081  chr19:16161672-16162168           NaN          NaN  NaN   \n",
       "13082  chr22:50511408-50511748           NaN          NaN  NaN   \n",
       "13086                      NaN           NaN          NaN  NaN   \n",
       "\n",
       "      bidirtesed_for_links  snp_in_gene  \n",
       "4                      NaN         True  \n",
       "7                      NaN         True  \n",
       "8                      NaN         True  \n",
       "9                    False         True  \n",
       "11                     NaN         True  \n",
       "...                    ...          ...  \n",
       "13079                  NaN         True  \n",
       "13080                  NaN         True  \n",
       "13081                False         True  \n",
       "13082                False         True  \n",
       "13086                  NaN         True  \n",
       "\n",
       "[5395 rows x 24 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ingenesnpsdf = allsnps_one_tissue[allsnps_one_tissue[\"closest_distance\"]==0]\n",
    "notgenesnpsdf = allsnps_one_tissue[allsnps_one_tissue[\"closest_distance\"]!=0]\n",
    "print (\"unique snps in genes\", ingenesnpsdf[\"rsID\"].nunique())\n",
    "print (\"unique snps not in genes\", notgenesnpsdf[\"rsID\"].nunique())\n",
    "ingenesnpsdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-11-28T22:29:05.389359Z",
     "start_time": "2023-11-28T22:29:05.381398Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Unnamed: 0', 'riskAllele', 'locations', 'snp_chr', 'snp_start', 'rsID',\n",
       "       'change', 'stop', 'snp_stop', 'closest_gene_chr', 'closest_gene_start',\n",
       "       'closest_gene_stop', 'closest_gene_name', 'score', 'strand',\n",
       "       'closest_distance', 'closest_gene', 'closest_transcript_id',\n",
       "       'bidirname', 'distance_tss', 'transcript_1', 'gene',\n",
       "       'bidirtesed_for_links', 'snp_in_gene'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "allsnps_one_tissue.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-11-28T22:41:46.994707Z",
     "start_time": "2023-11-28T22:41:46.921171Z"
    }
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Output what we know about all snps in this GWAS catalog"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-11-28T22:42:28.597938Z",
     "start_time": "2023-11-28T22:42:28.019766Z"
    }
   },
   "outputs": [],
   "source": [
    "\n",
    "def informationforpub(tissue, backtoallsnps):\n",
    "    print (\"riskAlleles\", len(backtoallsnps[\"riskAllele\"].unique()))\n",
    "    print (\"snp locations\", len(backtoallsnps[\"locations\"].unique()))\n",
    "    print (\"rsID\", len(backtoallsnps[\"rsID\"].unique()))\n",
    "    print (\"closest genes\", len(backtoallsnps[\"closest_gene\"].unique()))\n",
    "    ingenesnpsdf = backtoallsnps[backtoallsnps[\"closest_distance\"]==0]\n",
    "    print (\"snps in genes\", ingenesnpsdf[\"rsID\"].nunique())\n",
    "    print (\"closest genes when snp is in a gene\", len(ingenesnpsdf[\"closest_gene\"].unique()))\n",
    "    notingenesnpsdf = backtoallsnps[backtoallsnps[\"closest_distance\"]!=0]\n",
    "    print (\"snps not in  genes\", notingenesnpsdf[\"rsID\"].nunique())\n",
    "    print (\"closest genes when snp is not in a gene\", len(notingenesnpsdf[\"closest_gene\"].unique()))\n",
    "    \n",
    "    inabidir = backtoallsnps[~backtoallsnps[\"bidirname\"].isna()]\n",
    "    print (\"snps in any birectional\", inabidir[\"rsID\"].nunique())\n",
    "    inabidirnotgene = notingenesnpsdf[~notingenesnpsdf[\"bidirname\"].isna()]\n",
    "    inabidirinagene = ingenesnpsdf[~ingenesnpsdf[\"bidirname\"].isna()]\n",
    "    inabidirinagenetestedforsnps = backtoallsnps[backtoallsnps[\"bidirtesed_for_links\"]==True]\n",
    "    inabidirinagenetestedforsnps2 = inabidirnotgene[inabidirnotgene[\"bidirtesed_for_links\"]==True]\n",
    "    inabidirinagenetestedforsnps3 = inabidirinagene[inabidirinagene[\"bidirtesed_for_links\"]==True]\n",
    "    print (\"snps in any birectional that's not in a gene\", inabidirnotgene[\"rsID\"].nunique())\n",
    "    print (\"snps in any birectional that's in a gene\", inabidirinagene[\"rsID\"].nunique())\n",
    "    #/scratch/Shares/dowell/rutendo/projects/DBNascent_Analysis/data/counts/normalized/genes_inter_bidir_filtered.tsv.gz\n",
    "    print (\"snps in any birectional that was tested for correlations\", inabidirinagenetestedforsnps[\"rsID\"].nunique())\n",
    "    print (\"snps in any birectional that was tested for correlations (snp not in gene)\", inabidirinagenetestedforsnps2[\"rsID\"].nunique())\n",
    "    print (\"snps in any birectional that was tested for correlations (snp in gene)\", inabidirinagenetestedforsnps3[\"rsID\"].nunique())  \n",
    "    print (\"closeset genes when snp in any birectional\", len(inabidirnotgene[\"closest_gene\"].unique()))\n",
    "    inabidirlinkedtoagene = inabidirnotgene[~inabidirnotgene[\"transcript_1\"].isna()]\n",
    "    print (\"snps in any birectional and snp linked to gene\", inabidirlinkedtoagene[\"rsID\"].nunique())\n",
    "    print (\"closeset genes when snp in any birectional and snp linked to gene\", len(inabidirlinkedtoagene[\"closest_gene\"].unique()))\n",
    "    print (\"nascentlinked genes when snp in any birectional and snp linked to gene\", len(inabidirlinkedtoagene[\"transcript_1\"].unique()))\n",
    "    inabidirlinkedtoagene = inabidirnotgene[~inabidirnotgene[\"transcript_1\"].isna()]\n",
    "    repogene = [gene for gene in backtoallsnps[\"gene\"].unique() if gene not in backtoallsnps[\"closest_gene\"].unique()]\n",
    "    closestgene = [gene for gene in backtoallsnps[\"closest_gene\"].unique() if gene not in backtoallsnps[\"gene\"].unique()]\n",
    "    both = [gene for gene in backtoallsnps[\"closest_gene\"].unique() if gene in backtoallsnps[\"gene\"].unique()]\n",
    "    print (\"for all snps intergentic and geneic\")\n",
    "    print (\"\\tgenes that are nascentlinked and are not closeset genes\", len(repogene))\n",
    "    print (\"\\tgenes that are both in closeset genes and nascentlinked\", len(both))\n",
    "    print (\"\\tgenes that are closeset genes not nascentlinked\", len(closestgene))\n",
    "    repogene = [gene for gene in notingenesnpsdf[\"gene\"].unique() if gene not in notingenesnpsdf[\"closest_gene\"].unique()]\n",
    "    closestgene = [gene for gene in notingenesnpsdf[\"closest_gene\"].unique() if gene not in notingenesnpsdf[\"gene\"].unique()]\n",
    "    both = [gene for gene in notingenesnpsdf[\"closest_gene\"].unique() if gene in notingenesnpsdf[\"gene\"].unique()]\n",
    "    print (\"for snps that are intergentic\")\n",
    "    print (\"\\tgenes that are nascentlinked and are not closeset genes\", len(repogene))\n",
    "    print (\"\\tgenes that are both in closeset genes and nascentlinked\", len(both))\n",
    "    print (\"\\tgenes that are closeset genes not nascentlinked\", len(closestgene))\n",
    "    repogene = [gene for gene in inabidirnotgene[\"gene\"].unique() if gene not in inabidirnotgene[\"closest_gene\"].unique()]\n",
    "    closestgene = [gene for gene in inabidirnotgene[\"closest_gene\"].unique() if gene not in inabidirnotgene[\"gene\"].unique()]\n",
    "    both = [gene for gene in inabidirnotgene[\"closest_gene\"].unique() if gene in inabidirnotgene[\"gene\"].unique()]\n",
    "    print (\"for only snps in bidirs\")\n",
    "    print (\"\\tgenes that are nascentlinked and are not closeset genes\", len(repogene))\n",
    "    print (\"\\tgenes that are both in closeset genes and nascentlinked\", len(both))\n",
    "    print (\"\\tgenes that are closeset genes not nascentlinked\", len(closestgene))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-11-28T22:26:41.704720Z",
     "start_time": "2023-11-28T22:26:41.685946Z"
    }
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-11-28T22:42:37.591296Z",
     "start_time": "2023-11-28T22:42:31.137134Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "riskAlleles 8513\n",
      "snp locations 7805\n",
      "rsID 7805\n",
      "closest genes 1386\n",
      "snps in genes 3245\n",
      "closest genes when snp is in a gene 676\n",
      "snps not in  genes 4560\n",
      "closest genes when snp is not in a gene 949\n",
      "snps in any birectional 1061\n",
      "snps in any birectional that's not in a gene 579\n",
      "snps in any birectional that's in a gene 482\n",
      "snps in any birectional that was tested for correlations 393\n",
      "snps in any birectional that was tested for correlations (snp not in gene) 391\n",
      "snps in any birectional that was tested for correlations (snp in gene) 2\n",
      "closeset genes when snp in any birectional 349\n",
      "snps in any birectional and snp linked to gene 126\n",
      "closeset genes when snp in any birectional and snp linked to gene 99\n",
      "nascentlinked genes when snp in any birectional and snp linked to gene 259\n",
      "for all snps intergentic and geneic\n",
      "\tgenes that are nascentlinked and are not closeset genes 217\n",
      "\tgenes that are both in closeset genes and nascentlinked 43\n",
      "\tgenes that are closeset genes not nascentlinked 1343\n",
      "for snps that are intergentic\n",
      "\tgenes that are nascentlinked and are not closeset genes 221\n",
      "\tgenes that are both in closeset genes and nascentlinked 39\n",
      "\tgenes that are closeset genes not nascentlinked 910\n",
      "for only snps in bidirs\n",
      "\tgenes that are nascentlinked and are not closeset genes 234\n",
      "\tgenes that are both in closeset genes and nascentlinked 26\n",
      "\tgenes that are closeset genes not nascentlinked 323\n"
     ]
    }
   ],
   "source": [
    "informationforpub(tissue, allsnps_one_tissue)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# output about genes that have a snp in them and have a snp in a bidireciontl linked to them"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-11-28T22:42:37.612214Z",
     "start_time": "2023-11-28T22:42:37.595172Z"
    }
   },
   "outputs": [],
   "source": [
    "inabidirnotgene = allsnps_one_tissue[~allsnps_one_tissue[\"bidirname\"].isna()]\n",
    "inabidirnotgene = inabidirnotgene[~inabidirnotgene[\"transcript_1\"].isna()]\n",
    "inabidirnotgene = inabidirnotgene[inabidirnotgene[\"closest_distance\"]!=0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-11-28T22:42:41.186868Z",
     "start_time": "2023-11-28T22:42:37.615916Z"
    }
   },
   "outputs": [],
   "source": [
    "repogene = [gene for gene in inabidirnotgene[\"gene\"].unique() if gene not in inabidirnotgene[\"closest_gene\"].unique()]\n",
    "closestgene = [gene for gene in allsnps_one_tissue[\"closest_gene\"].unique() if gene not in allsnps_one_tissue[\"gene\"].unique()]\n",
    "both = [gene for gene in allsnps_one_tissue[\"closest_gene\"].unique() if gene in allsnps_one_tissue[\"gene\"].unique()]\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-11-28T22:42:41.200998Z",
     "start_time": "2023-11-28T22:42:41.190769Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "236"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(repogene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-11-28T22:42:41.217448Z",
     "start_time": "2023-11-28T22:42:41.210782Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "43"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(both)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-11-28T22:42:41.229286Z",
     "start_time": "2023-11-28T22:42:41.222934Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1343"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(closestgene)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# A intergenic snp we say regulates a gene and that gene has already got another snp inside it that is said to affect this disorder"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-11-28T22:42:41.628060Z",
     "start_time": "2023-11-28T22:42:41.233256Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['LINC00977',\n",
       " 'IKZF1',\n",
       " 'MDS2',\n",
       " 'MIR8059',\n",
       " 'FCRL3',\n",
       " 'GOLGA2P11',\n",
       " 'RGS2-AS1',\n",
       " 'UCHL5',\n",
       " 'ASB2',\n",
       " 'CDKN2A-DT',\n",
       " 'CDKN2A',\n",
       " 'CDKN2B-AS1',\n",
       " 'IFNA5',\n",
       " 'RCL1',\n",
       " 'PLXNA2',\n",
       " 'LNC-LBCS',\n",
       " 'SIM1',\n",
       " 'AGR2',\n",
       " 'LINC02227',\n",
       " 'TLX3',\n",
       " 'LINC01121',\n",
       " 'LOC100129434',\n",
       " 'RFPL1S',\n",
       " 'LINC02211',\n",
       " 'LINC01053',\n",
       " 'LINC00578',\n",
       " 'ABTB2',\n",
       " 'ASRGL1',\n",
       " 'HMX3',\n",
       " 'WTAPP1',\n",
       " 'IL15RA',\n",
       " 'PRKCQ',\n",
       " 'SFMBT2',\n",
       " 'MAP1LC3B2',\n",
       " 'MED13L',\n",
       " 'TMEM132C',\n",
       " 'AFF3',\n",
       " 'MKRN7P',\n",
       " 'LINC02313',\n",
       " 'TMEM260',\n",
       " 'SLC24A3',\n",
       " 'BALR6',\n",
       " 'LNX1-AS2']"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "repogene_ingenetoo = [gene for gene in inabidirnotgene[\"gene\"].unique() if gene in allsnps_one_tissue[\"closest_gene\"].unique()]\n",
    "repogene_ingenetoo\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-11-28T22:42:41.638622Z",
     "start_time": "2023-11-28T22:42:41.631805Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "43"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(repogene_ingenetoo)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-11-28T22:42:42.071243Z",
     "start_time": "2023-11-28T22:42:41.642671Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "LINC00977 did closest to this snp catch this gene too? True\n",
      "LINC00977 did closest to any snp catch this gene too? True\n",
      "IKZF1 did closest to this snp catch this gene too? True\n",
      "IKZF1 did closest to any snp catch this gene too? True\n",
      "MDS2 did closest to this snp catch this gene too? True\n",
      "MDS2 did closest to any snp catch this gene too? True\n",
      "MIR8059 did closest to this snp catch this gene too? True\n",
      "MIR8059 did closest to any snp catch this gene too? True\n",
      "FCRL3 did closest to this snp catch this gene too? True\n",
      "FCRL3 did closest to any snp catch this gene too? True\n",
      "GOLGA2P11 did closest to this snp catch this gene too? True\n",
      "GOLGA2P11 did closest to any snp catch this gene too? True\n",
      "RGS2-AS1 did closest to this snp catch this gene too? True\n",
      "RGS2-AS1 did closest to any snp catch this gene too? True\n",
      "UCHL5 did closest to this snp catch this gene too? False\n",
      "UCHL5 did closest to any snp catch this gene too? True\n",
      "ASB2 did closest to this snp catch this gene too? True\n",
      "ASB2 did closest to any snp catch this gene too? True\n",
      "CDKN2A-DT did closest to this snp catch this gene too? False\n",
      "CDKN2A-DT did closest to any snp catch this gene too? True\n",
      "CDKN2A did closest to this snp catch this gene too? False\n",
      "CDKN2A did closest to any snp catch this gene too? True\n",
      "CDKN2B-AS1 did closest to this snp catch this gene too? False\n",
      "CDKN2B-AS1 did closest to any snp catch this gene too? True\n",
      "IFNA5 did closest to this snp catch this gene too? True\n",
      "IFNA5 did closest to any snp catch this gene too? True\n",
      "RCL1 did closest to this snp catch this gene too? True\n",
      "RCL1 did closest to any snp catch this gene too? True\n",
      "PLXNA2 did closest to this snp catch this gene too? False\n",
      "PLXNA2 did closest to any snp catch this gene too? True\n",
      "LNC-LBCS did closest to this snp catch this gene too? True\n",
      "LNC-LBCS did closest to any snp catch this gene too? True\n",
      "SIM1 did closest to this snp catch this gene too? True\n",
      "SIM1 did closest to any snp catch this gene too? True\n",
      "AGR2 did closest to this snp catch this gene too? True\n",
      "AGR2 did closest to any snp catch this gene too? True\n",
      "LINC02227 did closest to this snp catch this gene too? False\n",
      "LINC02227 did closest to any snp catch this gene too? True\n",
      "TLX3 did closest to this snp catch this gene too? False\n",
      "TLX3 did closest to any snp catch this gene too? True\n",
      "LINC01121 did closest to this snp catch this gene too? True\n",
      "LINC01121 did closest to any snp catch this gene too? True\n",
      "LOC100129434 did closest to this snp catch this gene too? True\n",
      "LOC100129434 did closest to any snp catch this gene too? True\n",
      "RFPL1S did closest to this snp catch this gene too? False\n",
      "RFPL1S did closest to any snp catch this gene too? True\n",
      "LINC02211 did closest to this snp catch this gene too? True\n",
      "LINC02211 did closest to any snp catch this gene too? True\n",
      "LINC01053 did closest to this snp catch this gene too? False\n",
      "LINC01053 did closest to any snp catch this gene too? True\n",
      "LINC00578 did closest to this snp catch this gene too? True\n",
      "LINC00578 did closest to any snp catch this gene too? True\n",
      "ABTB2 did closest to this snp catch this gene too? True\n",
      "ABTB2 did closest to any snp catch this gene too? True\n",
      "ASRGL1 did closest to this snp catch this gene too? True\n",
      "ASRGL1 did closest to any snp catch this gene too? True\n",
      "HMX3 did closest to this snp catch this gene too? False\n",
      "HMX3 did closest to any snp catch this gene too? True\n",
      "WTAPP1 did closest to this snp catch this gene too? False\n",
      "WTAPP1 did closest to any snp catch this gene too? False\n",
      "IL15RA did closest to this snp catch this gene too? False\n",
      "IL15RA did closest to any snp catch this gene too? True\n",
      "PRKCQ did closest to this snp catch this gene too? False\n",
      "PRKCQ did closest to any snp catch this gene too? False\n",
      "SFMBT2 did closest to this snp catch this gene too? True\n",
      "SFMBT2 did closest to any snp catch this gene too? True\n",
      "MAP1LC3B2 did closest to this snp catch this gene too? False\n",
      "MAP1LC3B2 did closest to any snp catch this gene too? False\n",
      "MED13L did closest to this snp catch this gene too? True\n",
      "MED13L did closest to any snp catch this gene too? True\n",
      "TMEM132C did closest to this snp catch this gene too? False\n",
      "TMEM132C did closest to any snp catch this gene too? True\n",
      "AFF3 did closest to this snp catch this gene too? False\n",
      "AFF3 did closest to any snp catch this gene too? False\n",
      "MKRN7P did closest to this snp catch this gene too? True\n",
      "MKRN7P did closest to any snp catch this gene too? True\n",
      "LINC02313 did closest to this snp catch this gene too? False\n",
      "LINC02313 did closest to any snp catch this gene too? True\n",
      "TMEM260 did closest to this snp catch this gene too? False\n",
      "TMEM260 did closest to any snp catch this gene too? True\n",
      "SLC24A3 did closest to this snp catch this gene too? False\n",
      "SLC24A3 did closest to any snp catch this gene too? True\n",
      "BALR6 did closest to this snp catch this gene too? True\n",
      "BALR6 did closest to any snp catch this gene too? True\n",
      "LNX1-AS2 did closest to this snp catch this gene too? False\n",
      "LNX1-AS2 did closest to any snp catch this gene too? True\n",
      "    \n",
      "closest to this snp catch this gene too\n",
      "True 23\n",
      "False 20\n",
      "closest to any snp catch this gene too\n",
      "True 39\n",
      "False 4\n"
     ]
    }
   ],
   "source": [
    "closestgenecaugthitT=0\n",
    "closestgenecaugthitF=0\n",
    "closesttothissnpcaugthitT=0\n",
    "closesttothissnpcaugthitF=0\n",
    "for thisgene in repogene_ingenetoo:\n",
    "    onegeneintergenicallinfo=notgenesnpsdf[notgenesnpsdf[\"gene\"]==thisgene]\n",
    "    onegenegenicallinfo=onegeneintergenicallinfo[onegeneintergenicallinfo[\"closest_gene\"]==thisgene]\n",
    "    closesttothissnpcaugthit = len([agene for agene in onegeneintergenicallinfo[\"gene\"].unique() if agene in onegenegenicallinfo[\"closest_gene\"].unique()])==1\n",
    "    closestgenecaugthit = len([agene for agene in onegeneintergenicallinfo[\"gene\"].unique() if agene in notgenesnpsdf[\"closest_gene\"].unique()])==1\n",
    "    if closestgenecaugthit:\n",
    "        closestgenecaugthitT=closestgenecaugthitT+1\n",
    "    else:\n",
    "        closestgenecaugthitF=closestgenecaugthitF+1\n",
    "    if closesttothissnpcaugthit:\n",
    "        closesttothissnpcaugthitT=closesttothissnpcaugthitT+1\n",
    "    else:\n",
    "        closesttothissnpcaugthitF=closesttothissnpcaugthitF+1\n",
    "\n",
    "    print (thisgene, \"did closest to this snp catch this gene too?\", closesttothissnpcaugthit)\n",
    "    print (thisgene, \"did closest to any snp catch this gene too?\", closestgenecaugthit)\n",
    "    #print(\"      snp intergenic closest_gene gene (snp to gene distance)\", onegeneintergenicallinfo[[\"snp_start\", \"closest_gene\", \"closest_distance\"]])#\n",
    "    #print(\"      snp intergenic repo predicted gene (enhancer mu to gene distance)\", onegeneintergenicallinfo[[\"snp_start\", \"gene\", \"distance_tss\"]])#[\"snp_start\", \"gene\", \"distance_tss\"]\n",
    "    #print(\"      snp gene internal to gene\", onegenegenicallinfo[[\"snp_start\", \"closest_gene\", \"closest_distance\"]])#[\"snp_start\", \"closest_gene\", \"closest_distance\"]\n",
    "    \n",
    "    \n",
    "print (\"    \")    \n",
    "print (\"closest to this snp catch this gene too\")    \n",
    "print (\"True\", closesttothissnpcaugthitT)\n",
    "print (\"False\", closesttothissnpcaugthitF)\n",
    "print (\"closest to any snp catch this gene too\")    \n",
    "print (\"True\", closestgenecaugthitT)\n",
    "print (\"False\", closestgenecaugthitF)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2023-11-28T22:42:42.100525Z",
     "start_time": "2023-11-28T22:42:42.074971Z"
    }
   },
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'backtoallsnps' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-32-7bf23e815ebb>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mbacktoallsnps\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'backtoallsnps' is not defined"
     ]
    }
   ],
   "source": [
    "backtoallsnps.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
